### PR TITLE
fix: include server_name in the connection pool

### PR DESCRIPTION
**Pull request by Tieske**

### CODE REVIEW

**Comment by rainingmaster** [comment]
@Tieske Hi, I think we have the arg `pool`, and we can set the `pool` as `[<server_name>:]<host>:<port>` outside the lib. Could you describe a situation we need this special arg?

**Comment by Tieske** [comment]
the default might reuse a connection opened with different properties, and hence is a security issue

**Comment by Tieske** [comment]
probably this isn't even enough, see https://github.com/ledgetech/lua-resty-http/pull/206/files#diff-654e4e7b65177f9d0e6eb1d4b1c259dbR133-R143

File: lib/resty/redis.lua [hunk 1]
```diff
@@ -138,10 +138,16 @@ function _M.connect(self, host, port_or_opts, opts)
     local ok, err
 
     if unix then
-        ok, err = sock:connect(host, port_or_opts)
         opts = port_or_opts
+        if opts and opts.server_name and not opts.pool then
+            opts.pool = opts.server_name .. ":" .. host
```
Comment (rainingmaster) [inline lib/resty/redis.lua:143]: Should we follow [http](https://github.com/ledgetech/lua-resty-http/pull/206/files#diff-654e4e7b65177f9d0e6eb1d4b1c259dbR133-R143) module, put `opt.backlog`, `opt.ssl` and `opt.ssl_verify` into pool name too?

**Pull request closed without merging**